flate/inflate.go

```diff
@@ -379,7 +390,24 @@ func (f *decompressor) Read(b []byte) (int, error) {
 		if f.err != nil {
 			return 0, f.err
 		}
-		f.step(f)
+
+		switch f.step {
```

Comment:
Let's put this into a `func (f *decompressor) execStep()` to avoid the duplication.